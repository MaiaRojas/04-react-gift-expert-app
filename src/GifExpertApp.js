import React, {useState} from 'react';
import { AddCategory } from './componets/AddCategory';
import { GifGrid } from './componets/GifGrid';



export const GifExpertApp = ({ defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return <>
    <h2>GifExpertApp</h2>
    <hr />
    <AddCategory categories={categories} setCategories={setCategories} />

    <ol>
      {categories.map(category => <GifGrid key={category} category={category} />)}
    </ol>
  </>
};
