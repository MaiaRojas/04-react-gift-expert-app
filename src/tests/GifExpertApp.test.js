import React from 'react'
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Test <GifExpertApp />', () => {

  test('should be displayed correctly', () => {

    const wrapper = shallow(<GifExpertApp />)
    
    expect(wrapper).toMatchSnapshot();
  });
  

  test('should display a list of categories', () => {
    const categories = ['HunterxHunter', 'One piece'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper.find('GifGrid').length).toBe(categories.length)

  });
  
});
