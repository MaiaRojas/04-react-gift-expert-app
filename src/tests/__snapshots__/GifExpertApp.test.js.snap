// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> should be displayed correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
